perspective.el

```diff
@@ -1021,6 +1022,13 @@ perspective beginning with the given letter."
   (setq persp-show-modestring t)
   (persp-update-modestring))
 
+(defun persp-other-buffer (&optional buffer visible-ok frame)
+  "A version of `other-buffer' which respects perspectives."
+  (let ((candidates (persp-current-buffers)))
+    (do ((buf (other-buffer buffer visible-ok frame)
```

Comment:
Minor problem: this should be `cl-do` for consistency with the rest of the code and to keep the byte-compiler happy. We had to standardize use of `loop` to `cl-loop` in the past for these reasons.

Bigger problem: this code started an infinite loop when I called it in a new perspective.